c10/core/MemoryFormat.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <c10/core/Backend.h>
+#include <c10/util/Exception.h>
+
+#include <iostream>
+
+// Memory format is not the property of a Tensor. It is the way to tell an
+// operator how the result should be organized in memory and nothing more. That
+// means memory format should never be used as return value for any tensor state
+// interrogation functions (internally and externally).
```

Comment:
But this isn't 100% true, because we *do* actually store the tag inside Tensor. Can we expand this documentation to explain the meaning of this?